plan/plan_test.go

```diff
@@ -357,3 +372,59 @@ func validateEntries(t *testing.T, entries, expected []*endpoint.Endpoint) {
 		t.Fatalf("expected %q to match %q", entries, expected)
 	}
 }
+
+func TestValidateDNSName(t *testing.T) {
+
+	records := []struct {
+		dnsName string
+		expect  string
+		err     error
+	}{
+		{
```

Comment:
can you add test cases for edge cases, e.g. a domain name without any dot?

Reply:
That was done.